docs/DeveloperGuide.md

```diff
@@ -4,26 +4,207 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+# 1. Overview
+MeetingJio is a program for **finding potential time slots for team meetings based on everyone’s availability**.
+
+# 2. Design 
+
+## 2.1 Architecture
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## 2.2 UI Component
+
+## 2.3 Logic Component
+- Parsers
+- Commands
+
+## 2.4 Model Component
+- Timetables
+- Events
+
+## 2.5 Storage Component
+
+# 3. Implementation
+This section describes some noteworthy details on how certain features are implemented.
+
+## 3.1 Add Feature
+The `add` command is a command that allows user to 
+1. add new users to the master timetable.
+2. add new events to the timetables.
+
+The following sequence diagram shows how the `add` operation generally works.
+
+![AddCommandSequenceDiagram](diagrams/AddCommandSequenceDiagram.png)
```

Comment:
Should parseCommand() call be a solid line?
![image](https://user-images.githubusercontent.com/91527581/161181101-c6021c7f-7969-463f-a2e2-784798a2a6b9.png)
Perhaps the deletion cross should end the life line, rather than the activation bar?
![image](https://user-images.githubusercontent.com/91527581/161181227-6540fe6e-bd54-4de7-9a58-242b0295726e.png)
E.g.
![image](https://user-images.githubusercontent.com/91527581/161181327-6a46e7df-b522-4db3-9a04-58e0b81bad0e.png)